manifests/params.pp

```diff
@@ -34,7 +34,24 @@
   $postrun_command     = undef
   $dns_alt_names       = []
   $use_srv_records     = false
-  $srv_domain          = $::domain
+
+  # On versions < 3.7.5 there was no defined function as such strict variables cannot be used
+  # On verisons < 4.0.0 and >= 3.5.0 the facts hash was optional disabled by default, on version < 3.5.0 the facts hash did not exist
+  # In some edge cases 'domain' fact may be not defined
+  if versioncmp($::puppetversion, '3.7.5') < 0 {
+    $srv_domain = $::domain
+  } elsif versioncmp($::puppetversion, '4.0.0') >= 0 {
```

Comment:
Right, but this comment is about the PR here, not the code in master. This block for the 4.0.0 version test is probably unnecessary, the `elsif defined(..)` test should be sufficient.